pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -211,6 +228,14 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 			return reconcile.Result{}, err
 		}
 
+		// do not fail immediatlty, but try again if the method fails because of the update conflict
+		if err = r.retryCheckAndDecrement(machine); err != nil {
```

Comment:
I'm confused. It's difficult to say what this function is meant to do by the name but it seems to be meant to update and decrement the number of disruptions allowed, but the remediation is not decided to take place until line 240 so why trying to decrease here?

Reply:
The good point moved under `if unhealthyForTooLong(nodeCondition, conditionTimeout)`, so it will decrease the `MachineDisruptionsAllowed` only just before the remediation of the machine and also updated the name of the method.